### PR TITLE
enchanced events details and edit views

### PR DESCRIPTION
**Pull request by iverhun**

### CODE REVIEW

File: src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.vaadin.data.HasValue;
+import com.vaadin.shared.Registration;
+import com.vaadin.shared.ui.ContentMode;
+import com.vaadin.ui.Label;
+import org.apache.commons.lang3.StringUtils;
+
+import static com.lohika.morning.ecs.utils.EcsUtils.formatString;
+
+public class EcsLabel extends Label implements HasValue<String> {
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java]: By the name of the class it is unclear what he is doing

File: src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    this.isLink = isLink;
+  }
+
+  @Override
+  public void setValue(String value) {
+    String linkValue = value;
+
+    if (StringUtils.isNotBlank(value) && isLink) {
+      setContentMode(ContentMode.HTML);
+      linkValue = formatString("<a href=\"{}\">{}</a>", value, value);
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java]: What about single responsibility?

File: src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java (hunk 1 above)
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java]: Do we needed `implements HasValue<String> ` ?

File: src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  @Override
+  public void setReadOnly(boolean readOnly) {
+    throw new UnsupportedOperationException("Label is read only by definition");
+  }
+
+  @Override
+  public boolean isReadOnly() {
+    return true;
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java]: can we keep parent behavior?

File: src/main/java/com/lohika/morning/ecs/domain/event/EventDetailsView.java [hunk 4]
```diff
@@ -32,34 +34,43 @@
...
-    private Button importTalksBtn = new Button("Import Talks and Speakers", VaadinIcons.ARROW_DOWN);
+    private Button importTalksBtn = new Button("Import Talks and Speakers", VaadinIcons.DOWNLOAD);
     private HorizontalLayout actions = new HorizontalLayout(editBtn, deleteBtn, closeBtn);
 
     private TalksList talksList;
-    private final FormLayout eventDetails = new FormLayout();
+    private final FormLayout eventDetails;
     private final VerticalLayout talksPanel;
 
+    private Binder<MorningEvent> binder = new Binder<>(MorningEvent.class);
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/domain/event/EventDetailsView.java]: Do we needed `binder` on details view page?

File: src/main/java/com/lohika/morning/ecs/domain/event/EventEditorView.java [hunk 5]
```diff
@@ -45,11 +45,11 @@
     /* Action buttons */
     private Button saveBtn = new Button("Save", VaadinIcons.CHECK);
     private Button cancelBtn = new Button("Cancel");
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/domain/event/EventEditorView.java]: on `cancel` we should go to view page when event saved.

File: src/main/java/com/lohika/morning/ecs/domain/event/EventDetailsView.java [hunk 6]
```diff
@@ -32,34 +34,43 @@
...
+    private EcsLabel description = new EcsLabel("Description");
+    private EcsLabel date = new EcsLabel("Date");
+    private EcsLabel ticketsUrl = new EcsLabel("Tickets URL", true);
 
     /* Action buttons */
     private Button editBtn = new Button("Edit", VaadinIcons.EDIT);
     private Button closeBtn = new Button("Close", VaadinIcons.CLOSE);
     private Button deleteBtn = new Button("Delete", VaadinIcons.TRASH);
-    private Button importTalksBtn = new Button("Import Talks and Speakers", VaadinIcons.ARROW_DOWN);
+    private Button importTalksBtn = new Button("Import Talks and Speakers", VaadinIcons.DOWNLOAD);
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/domain/event/EventDetailsView.java]: do we need import here?

File: src/main/java/com/lohika/morning/ecs/domain/event/EventEditorView.java [hunk 7]
```diff
@@ -94,8 +99,10 @@ public void enter(ViewChangeListener.ViewChangeEvent viewChangeEvent) {
 
         // add listeners
         saveBtn.addClickListener(clickEvent -> {
-            eventService.save(morningEvent);
-            navigateTo(EventDetailsView.VIEW_NAME, morningEvent.getEventNumber());
+            if (binder.validate().isOk()) {
```
Comment (banadiga) [inline src/main/java/com/lohika/morning/ecs/domain/event/EventEditorView.java]: I prefer to extract functionality to separate method

File: src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java (hunk 1 above)
Comment (iverhun) [reply src/main/java/com/lohika/morning/ecs/vaadin/EcsLabel.java]: Just a custom (ECS) label implementation

**Pull request merged**